### PR TITLE
WIP: FW offboard control

### PR DESCRIPTION
**Pull request by AndreasAntener**
**WIP**, need to do new tests to verify this is working. I moved the offboard waypoint logic we used last week from the mavlink receiver to the fw position controller since I think it makes more sense there because of the way it works. (will squash the commits once tested)

Basically works as follows:
- reads offboard position setpoints
- uses latest setpoint as target
- uses previous setpoint as starting point for the L1 line (updated if the current setpoint changes)

@LorenzMeier @tumbili Thoughts on the general approach? I can think of different offboard control modes for FW but in terms of position control this seems to be the most natural one.

### CODE REVIEW

**Comment by tumbili** [comment]
@AndreasAntener I had a look at it and it seems fine. I wouldn't know how to implement it any better.

**Comment by AndreasAntener** [comment]
Tested with a standard VTOL. Here's a flight log from today: http://dash.oznet.ch/view/PmdKn4VPrMbCAVC4eQbBuN
A ROS python script was used to fly a square, which you see in the second half of the flight.

**Comment by AndreasAntener** [comment]
Here's another flight log, from a complete offboard flight including transition: http://dash.oznet.ch/view/q6UkmjK7Kpr3sLEoTf3JPn
Used #2670 to command the transition

File: src/modules/fw_pos_control_l1/fw_pos_control_l1_main.cpp [hunk 1]
```diff
@@ -1616,6 +1630,111 @@ FixedwingPositionControl::control_position(const math::Vector<2> &current_positi
...
+				_mTecs.resetDerivatives(_airspeed.true_airspeed_m_s);
+			}
+		}
+		_control_mode_current = FW_POSCTRL_MODE_OFFBOARD;
+
+		/* current waypoint (the one we're currently heading for) */
+		math::Vector<2> curr_wp;
+		if (pos_sp_triplet.current.valid &&
+				pos_sp_triplet.current.position_valid) {
+			map_projection_reproject(&_home_ref,
```
Comment (bansiesta) [inline src/modules/fw_pos_control_l1/fw_pos_control_l1_main.cpp]: @LorenzMeier: Is it the task of the controllers to do all the projections? Aren't pos_sp_triplet.current.lat, etc. already provided?

Comment (AndreasAntener) [inline src/modules/fw_pos_control_l1/fw_pos_control_l1_main.cpp]: In offboard you only get what has been sent, for position control these are only x/y/z local coordinates.

Compared with the MC pos controller we have kind of the opposite here: the MC controller in its essence works with local coordinates, offboard position control uses the setpoint input directly. In `auto` it converts from lat/lon to local. But the navigator for FW works with lat/lon coordinates so no projection happens in `auto`.

I wouldn't say it is wrong to do projections in the position controllers. It needs to be done somewhere where you know the correct reference.

**Comment by AndreasAntener** [comment]
TODO: need to look at updating the local position reference again, maybe also check for reference updates during offboard like the MC pos controller does.

**Comment by AndreasAntener** [comment]
I have bigger plan for this now and will make a new PR once implemented.

**Pull request closed without merging**